app/assets/javascripts/components/AccountMenu/index.tsx

```diff
@@ -0,0 +1,163 @@
+
+  const { notesAndTagsCount } = appState.accountMenu;
+
+  const closeAccountMenu = () => {
+    appState.accountMenu.closeAccountMenu();
+  };
+
+  // Add the required event observers
+  useEffect(() => {
+    const removeKeyStatusChangedObserver = application.addEventObserver(
+      async () => {
+        setUser(application.getUser());
+      },
+      ApplicationEvent.KeyStatusChanged
+    );
+
+    const removeProtectionSessionExpiryDateChangedObserver = application.addEventObserver(
+      async () => {
+        setProtectionsDisabledUntil(getProtectionsDisabledUntil());
+      },
+      ApplicationEvent.ProtectionSessionExpiryDateChanged
+    );
+
+    return () => {
+      removeKeyStatusChangedObserver();
+      removeProtectionSessionExpiryDateChangedObserver();
+    };
+  }, [application, getProtectionsDisabledUntil]);
+
+  return (
+    <div className="sn-component">
+      <div id="account-panel" className="sk-panel">
+        <div className="sk-panel-header">
+          <div className="sk-panel-header-title">Account</div>
+          <a className="sk-a info close-button" onClick={closeAccountMenu}>Close</a>
+        </div>
+        <div className="sk-panel-content">
+          <Authentication
+            application={application}
+            server={server}
+            setServer={setServer}
+            closeAccountMenu={closeAccountMenu}
+            notesAndTagsCount={notesAndTagsCount}
+            showLogin={showLogin}
+            setShowLogin={setShowLogin}
+            showRegister={showRegister}
+            setShowRegister={setShowRegister}
+            user={user}
+          />
+          {!showLogin && !showRegister && (
+            <div>
+              {user && (
+                <User
+                  email={user.email}
+                  server={server}
+                  appState={appState}
+                  application={application}
+                  closeAccountMenu={closeAccountMenu}
+                />
+              )}
+              <Encryption
+                isEncryptionEnabled={isEncryptionEnabled}
+                notesAndTagsCount={notesAndTagsCount}
+                encryptionStatusString={encryptionStatusString}
+              />
+              {hasProtections && (
+                <Protections
+                  application={application}
+                  protectionsDisabledUntil={protectionsDisabledUntil}
+                />
+              )}
+              <PasscodeLock
+                application={application}
+                setEncryptionStatusString={setEncryptionStatusString}
+                setIsEncryptionEnabled={setIsEncryptionEnabled}
+                setIsBackupEncrypted={setIsBackupEncrypted}
+              />
+              <DataBackup
+                application={application}
+                isBackupEncrypted={isBackupEncrypted}
+                isEncryptionEnabled={isEncryptionEnabled}
+                setIsBackupEncrypted={setIsBackupEncrypted}
```

Comment:
To keep it cleaner we could move state variables that are shared between several components, such as isBackupEncrypted, isEncryptionEnabled, and their setters, to the shared appState using Mobx

Reply:
Good point, done!